Assets/Scripts/Models/Character.cs

```diff
@@ -1017,9 +999,9 @@ private bool CheckForJobMaterials()
 
     private bool WalkingToUsableInventory()
     {
-        bool destHasInventory = pathAStar != null && pathAStar.EndTile() != null && pathAStar.EndTile().Inventory != null;
+        bool destHasInventory = movementPath != null && movementPath.Last() != null && movementPath.Last().Inventory != null;
```

Comment:
I just did a test run and it looked like two characters (they were on top of each other) walked to pickup the same inventory item. When they reached the spot, one grabbed the inventory and walked away and the other stayed put (which is the only reason I could tell there were two there), and the log was spammed with
```
InvalidOperationException: Operation is not valid due to the current state of the object
System.Linq.Enumerable.Last[Tile] (IEnumerable`1 source)
Character.WalkingToUsableInventory () (at Assets/Scripts/Models/Character.cs:1002)
Character.CheckForJobMaterials () (at Assets/Scripts/Models/Character.cs:943)
Character.Update_DoJob (Single deltaTime) (at Assets/Scripts/Models/Character.cs:698)
Character.Update (Single deltaTime) (at Assets/Scripts/Models/Character.cs:336)
World.TickEveryFrame (Single deltaTime) (at Assets/Scripts/Models/World.cs:188)
TimeManager.InvokeEvent (System.Action`1 eventAction, Single time) (at Assets/Scripts/Controllers/TimeManager.cs:198)
TimeManager.Update (Single time) (at Assets/Scripts/Controllers/TimeManager.cs:131)
WorldController.Update () (at Assets/Scripts/Controllers/WorldController.cs:146)
```
each tick.